sts/client.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright (c) 2018 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sts
+
+import (
+	"context"
+	"crypto/tls"
+	"net/url"
+	"time"
+
+	"github.com/vmware/govmomi/sts/internal"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/soap"
+)
+
+// Client is a soap.Client targeting the STS (Secure Token Service) API endpoint.
+type Client struct {
+	*soap.Client
+}
+
+// DefaultPath is the default URL Path for the STS service
+var DefaultPath = "/sts/STSService"
+
+// NewClient returns an STS client targeting DefaultPath.
+func NewClient(ctx context.Context, c *vim25.Client) (*Client, error) {
+	sc := c.Client.NewServiceClient(DefaultPath, "urn:oasis:names:tc:SAML:2.0:assertion")
+
+	return &Client{sc}, nil
+}
+
+// TokenRequest parameters for issuing a SAML token.
+// At least one of Userinfo or Certificate must be specified.
+type TokenRequest struct {
+	Userinfo    *url.Userinfo    // Userinfo when set issues a Bearer token
+	Certificate *tls.Certificate // Certificate when set issues a HoK token
+	ActAs       string           // ActAs identity when set is delegated to the caller
+	Duration    time.Duration    // Duration is the token's lifetime, defaults to 1h
+	Renewable   bool             // Renewable allows the issued token to be renewed
+	Delegatable bool             // Delegatable allows the issued token to be delegated (e.g. for use with ActAs)
```

Comment:
Is `Delegatable` actually for use with `ActAs`? I can't find documentation that talks about that.

As far as I can tell, `Delegatable` is about allowing the issued token to be delegated to a party other than the one making the issue request, but `ActAs` indicates that the requestor itself will be using the token in order to act as another identify (but not necessarily delegating the issued token to another party). (It seems like the two together would allow for requesting a token to act as another identity and then delegating that token to a third party.)

In other words, it seems like `ActAs` is about who the token _represents_, but `Delegatable` is about who the token is _used by_.